docs: fix UI Swagger spec link  #1207

uday-codes69 opened the pull request:
This PR fixes Issue #1207 by uploading and verifying the OpenAPI YAML file using both Swagger UI and ReDoc. The spec renders correctly without errors, and all endpoints are visible as expected.

✅ Work Done
	•	Uploaded the YAML to ReDoc and confirmed proper rendering
	•	Verified the same on Swagger UI
	•	No schema errors or missing endpoints found

Fixes #1207 

	•	Added Swagger UI link in README.md pointing to swagger.yaml.
	•	Verified the OpenAPI spec renders correctly using ReDoc (see screenshot below).

<img width="774" height="410" alt="Screenshot 2025-07-30 at 10 47 56 PM" src="https://github.com/user-attachments/assets/a170b377-8253-4e37-8618-7e7e92ab5a9f" />

pboyd reviewed (approved):
/lgtm

tarilabs reviewed (changes requested):
thank you, but can't we just point users here?

https://www.kubeflow.org/docs/components/model-registry/reference/rest-api/#swagger-ui

uday-codes69 commented:
> thank you, but can't we just point users here?
> 
> https://www.kubeflow.org/docs/components/model-registry/reference/rest-api/#swagger-ui

Thank you for the feedback! That makes sense — I'll update the PR to just link to the official docs.

uday-codes69 commented:
Hi @tarilabs @lucferbux,  
I’ve added the UI Swagger API doc under `clients/ui/api/openapi/` and updated the links accordingly, following the structure of the main API spec.  
Please review and let me know if further adjustments are needed. Thanks for the guidance!

tarilabs commented on README.md:
```diff
@@ -204,3 +204,7 @@ manifests/kustomize/overlays/db/model-registry-db-deployment.yaml file.
+
+> The Swagger Editor no longer supports external URLs directly.  
+You can view the UI API spec using Redoc here:
+[View UI Swagger Spec (Redoc)](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/kubeflow/model-registry/main/clients/ui/api/openapi/mod-arch.yaml)
```
can we transform this from pointing users to 
https://www.kubeflow.org/docs/components/model-registry/reference/rest-api/#swagger-ui
please.

tarilabs replied:
☝️

uday-codes69 replied:
Thanks @tarilabs for pointing that out!  
I’ve updated the **root README.md** to link directly to the official Kubeflow docs (`rest-api/#swagger-ui`) instead of the Redoc/raw GitHub link.  
This should now match the approach used in the main API docs. Appreciate your guidance 🙏

tarilabs commented:
can you please sign commits @uday-codes69 ?
see guide: https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits

uday-codes69 commented:
> can you please sign commits @uday-codes69 ? see guide: https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits

Hi @tarilabs  I’ve updated the PR as requested — all commits are now signed off and the DCO check is passing ✅. Thanks for guiding me through this!

uday-codes69 commented:
Hi @lucferbux  just following up — the PR is now updated with all commits signed off ✅ as requested. Would really appreciate it if you could take a moment to review it when you get a chance. Thanks in advance!

bobbravo2 commented:
[This PR](https://github.com/kubeflow/model-registry/pull/1464) might be made obsolete by the one here. Great work @uday-codes69 !

tarilabs commented on README.md:
````diff
@@ -204,3 +204,12 @@ manifests/kustomize/overlays/db/model-registry-db-deployment.yaml file.
 spec.template.spec.containers.image: public.ecr.aws/docker/library/mysql:8.3.0
 ```
+
+> The Swagger Editor no longer supports external URLs directly.  
````
this PR is failing UI test?

ederign commented:
This PR is stale and partially fixed by https://github.com/kubeflow/model-registry/pull/1464 . @uday-codes69 feel free to reopen with an updated version!
